File: src/main/java/com/lessmarkup/interfaces/text/SearchResults.java
package com.lessmarkup.interfaces.text;

import java.util.ArrayList;
import java.util.List;

public class SearchResults {
    private final List<SearchResult> results = new ArrayList<>();
    private int actualCount;
    
    public List<SearchResult> getResults() {
        return this.results;
    }
    
    public int getActualCount() {
        return this.actualCount;
    }
    
    public void setActualCount(int val) {
        this.actualCount = val;
    }
}

File: src/main/scala/com/lessmarkup/interfaces/text/SearchResults.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.text

import scala.collection.mutable

class SearchResults {
  private final val results: mutable.ListBuffer[SearchResult] = mutable.ListBuffer()
  private var actualCount: Int = 0

  def getResults: List[SearchResult] = {
    this.results.toList
  }

  def getActualCount: Int = {
    this.actualCount
  }

  def setActualCount(`val`: Int) {
    this.actualCount = `val`
  }
}